google_api_availability_platform_interface/lib/src/google_api_availability_platform_interface.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:google_api_availability_platform_interface/google_api_availability_platform_interface.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'implementations/method_channel_google_api_availability.dart';
+
+/// The interface that implementations of `google_api_availability` must implement.
+///
+/// Platform implementations should extend this class rather than implement it
+/// as `google_api_availability` does not consider newly added methods to be
+/// breaking changes. Extending this class (using `extends`) ensures that the
+/// subclass will get the default implementation, while platform implementations
+/// that `implements` this interface will be broken by newly added
+/// [GoogleApiAvailabilityPlatform] methods.
+abstract class GoogleApiAvailabilityPlatform extends PlatformInterface {
+  /// Constructs a [GoogleApiAvailabilityPlatform].
+  GoogleApiAvailabilityPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static GoogleApiAvailabilityPlatform _instance =
+      MethodChannelGoogleApiAvailability();
+
+  /// The default instance of [GoogleApiAvailabilityPlatform] to use.
+  ///
+  /// Defaults to [MethodChannelGoogleApiAvailability].
+  static GoogleApiAvailabilityPlatform get instance => _instance;
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [GoogleApiAvailabilityPlatform] when they register
+  /// themselves.
+  static set instance(GoogleApiAvailabilityPlatform instance) {
+    PlatformInterface.verify(instance, _token);
+    _instance = instance;
+  }
```

Comment:
The use of a default implementation is no longer recommended as it increases maintenance for something that most likely will never be used. 

Current recommended solution is to make the `_instance` field `null`able. Something like this:

```suggestion
  static GoogleApiAvailabilityPlatform? _instance;

  /// The default instance of [GoogleApiAvailabilityPlatform] to use.
  ///
  /// Defaults to [MethodChannelGoogleApiAvailability].
  static GoogleApiAvailabilityPlatform? get instance => _instance;

  /// Platform-specific plugins should set this with their own platform-specific
  /// class that extends [GoogleApiAvailabilityPlatform] when they register
  /// themselves.
  static set instance(GoogleApiAvailabilityPlatform? instance) {
      if (instance == null) {
        throw AssertionError(
            'Platform interfaces can only be set to a non-null instance');
      }
  
    PlatformInterface.verify(instance, _token);
    _instance = instance;
  }
```